src/reducers/fileTree.spec.tsx

```diff
@@ -585,4 +589,417 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('getRelativeMessageUid', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const file3 = 'file3.js';
+    const line1 = 1;
+    const line2 = 2;
+    const line3 = 3;
+    const line1FirstUid = 'line1-uid-1';
+    const line1SecondUid = 'line1-uid-2';
+    const line2Uid = 'line2-uid';
+    const line3Uid = 'line3-uid';
+    const global1 = 'global-uid-1';
+    const global2 = 'global-uid-2';
+
+    const createMessageMap = (
```

Comment:
I think you have made more work for yourself than necessary with this helper. I'm pretty sure you can just use `getMessageMap()`, right? I tried replacing a few tests and it seemed to work:

```diff
diff --git a/src/reducers/fileTree.spec.tsx b/src/reducers/fileTree.spec.tsx
index 5e0f2b7..a002f1a 100644
--- a/src/reducers/fileTree.spec.tsx
+++ b/src/reducers/fileTree.spec.tsx
@@ -15,8 +15,13 @@ import reducer, {
   initialState,
 } from './fileTree';
 import { createInternalVersion, createInternalVersionEntry } from './versions';
-import { ExternalLinterMessage, LinterMessageMap } from './linter';
 import {
+  ExternalLinterMessage,
+  LinterMessageMap,
+  getMessageMap,
+} from './linter';
+import {
+  createFakeExternalLinterResult,
   createFakeLinterMessagesByPath,
   createFakeThunk,
   createVersionWithEntries,
@@ -640,7 +645,11 @@ describe(__filename, () => {
       expect(
         _getRelativeMessageUid({
           currentMessageUid: '',
-          messageMap: createMessageMap([]),
+          messageMap: getMessageMap(
+            createFakeExternalLinterResult({
+              messages: [],
+            }),
+          ),
         }),
       ).toEqual(null);
     });
@@ -649,9 +658,11 @@ describe(__filename, () => {
       const currentPath = 'currentPath.js';
       const messagePath = 'notTheCurrentPath.js';
 
-      const messageMap = createMessageMap([
-        { path: messagePath, messages: [{ line: line1, uid: 'line1' }] },
-      ]);
+      const messageMap = getMessageMap(
+        createFakeExternalLinterResult({
+          messages: [{ file: messagePath, line: line1, uid: 'line1' }],
+        }),
+      );
 
       expect(() => {
         _getRelativeMessageUid({
@@ -663,16 +674,15 @@ describe(__filename, () => {
     });
 
     it('throws an error with an empty pathList', () => {
-      const externalMessages1 = [
-        { line: null, uid: global1 },
-        { line: line1, uid: line1FirstUid },
-      ];
-      const externalMessages2 = [{ line: null, uid: global2 }];
-
-      const messageMap = createMessageMap([
-        { path: file1, messages: externalMessages1 },
-        { path: file3, messages: externalMessages2 },
-      ]);
+      const messageMap = getMessageMap(
+        createFakeExternalLinterResult({
+          messages: [
+            { file: file1, line: null, uid: global1 },
+            { file: file1, line: line1, uid: line1FirstUid },
+            { file: file3, line: null, uid: global2 },
+          ],
+        }),
+      );
 
       expect(() => {
         _getRelativeMessageUid({
```

Reply:
Yes, this is better. I think my way meant not having to specify a file name multiple times for multiple messages, but I agree that this is better.